src/Nokia_LCD.h

```diff
@@ -231,6 +231,14 @@ class Nokia_LCD {
      */
     void setDefaultFont();
 
+    /**
+     * couple()/uncouple() is useful for improving speed and 
+     * for accessing two or more displays on the same pins
+     * (but different CE pins)
+     */
```

Comment:
The claims are that using these two new functions:
1. Improve speed
2. Allow accessing more than 1 display on the same pins

I can understand the 2nd point, but can you elaborate more on the 1st?

Reply:
Two answer all your remarks and maybe more:

1. Why this proposal

I was experimenting with two displays. But they constantly resisted to work as expected. I suspect electrical interference between the CE lines. Another advantage is that the contrast setting menu can be shown on all displays the same time so if your application misconfigured one LCD but shows the contrast setting menu on the other LCD too you still can read what's going on.


2. How-to use

The key of couple() is to stop the send() method from modifying the CE pin. Obviously you have to manage it by yourself then.

The bonus of couple() is that you can switch from one display to another without changing all these lcd.print() invocations in main and subroutines. I use this - but to be frank it would be more clear to wrap a digitalWrite such as activate_lcd1() {digitalWrite(CEofLCD1, LOW);} - I probably will switch to this soon for sake of clarity.


3. Examples

See
a) Full http://z-info.jbechtel.de/srv/23kopierer_savev179web.ino
b) Reduced http://z-info.jbechtel.de/srv/23kopierer_web_short.ino
c) Mini http://z-info.jbechtel.de/srv/23kopierer_web_veryshort.ino

The a) example is an actually working one. But it contains lots of code which you don't need for LCD (and also lots of dead code).
The b) example is untested. I just took the a) example and removed everything which is not contrast setup menu/counter printer.
The c) example is pretty minimal, just has the initialization phase of b) example. Also untested.

Please don't mind the "lcd.uncouple()" in all a)b)c) examples initialization. It is wrong and most probably should be just removed.
Also please don't mind that it is not neccessary to instantiate lcdL and lcdR objects. lcd object is just enough. All lcdL and lcdR tasks could be surrounded by activate_lcd1() like wrappers (see above 2.).


4. Improve speed

It was just my feeling that on Arduino the PIN access is pretty slow, compared to directly coding for ATMega. Web search engine results support this. (digitalWrite() seems to switch off PWM first, etc.). So omitting digitalWrite(), even just the surrounding one for CE, helps a bit. Of course one could be pretty much faster when not-at-all using digitalWrite() but reducint calls is not so bad, either.


I have incorporated most of your suggestions and expanded the documentation comment in Nokia_LCD.h. Is it better than before?